templates/sfn/ingest_sfn_definition.json.tpl

```diff
@@ -201,23 +201,42 @@
           "archiveHierarchyFolders.$": "$.archiveHierarchyFolders",
           "contentFolders.$": "$.contentFolders"
         },
-        "contentAssets.$": "$.contentAssets"
+        "contentAssets.$": "$.contentAssets",
+        "assetBucket.$": "$.asset.bucket",
+        "assetKey.$": "$.asset.key",
+        "foldersBucket.$": "$.folders.bucket",
+        "foldersKey.$": "$.folders.key"
       }
     },
     "Merge array": {
       "Type": "Pass",
       "Next": "Map over each Folder Id",
       "Parameters": {
         "allFolders.$": "$.allFolders.[*]*",
+        "assetsBucket.$": "$.assetsBucket",
+        "assetsKey.$": "$.assetsKey",
         "contentAssets.$": "$.contentAssets",
         "executionId.$": "$$.Execution.Name",
+        "foldersBucket.$": "$.foldersBucket",
+        "foldersKey.$": "$.foldersKey",
         "workflowContextName": "Ingest OPEX (Incremental)",
         "stagingPrefix.$": "States.Format('opex/{}', $$.Execution.Name)",
         "stagingBucket": "${ingest_staging_cache_bucket_name}"
       }
     },
     "Map over each Folder Id": {
       "Type": "Map",
+      "Label": "MapOverEachFolderId",
+      "ItemReader": {
+        "Resource": "arn:aws:states:::s3:getObject",
+        "ReaderConfig": {
+          "InputType": "JSON"
+        },
+        "Parameters": {
+          "Bucket": "$.foldersBucket",
+          "Key.$": "$.foldersKey"
+        }
```

Comment:
See other comment, this should make these `$.folders.bucket` and `$.folders.key` respectively. Nice and simple :)